scopes/lanes/ui/hooks/lanes-context/lanes-model.ts

```diff
@@ -135,15 +135,23 @@ export class LanesModel {
     return grouped;
   }
 
-  static groupByComponentHash(lanes: LaneModel[]): Map<string, { lane: LaneModel; component: LaneComponentModel }> {
-    const grouped = new Map<string, { lane: LaneModel; component: LaneComponentModel }>();
+  static groupByComponentHashAndId(lanes: LaneModel[]): {
+    byHash: Map<string, { lane: LaneModel; component: LaneComponentModel }>;
+    byId: Map<string, LaneModel[]>;
+  } {
+    const byHash = new Map<string, { lane: LaneModel; component: LaneComponentModel }>();
+    const byId = new Map<string, LaneModel[]>();
     lanes.forEach((lane) => {
       const { components } = lane;
       components.forEach((component) => {
-        grouped.set(component.model.id.version as string, { lane, component });
+        const id = component.model.id.fullName;
+        const version = component.model.id.version as string;
+        byHash.set(version, { lane, component });
+        const existing = byId.has(id) ? (byId.get(id) as LaneModel[]) : [];
+        byId.set(id, [...existing, lane]);
```

Comment:
maybe `byId.get(id) || []` (saves the extra type cast)
and since this is a new array during construction, you can use `.push`